### DISCUSSION TITLE
Note title selected as opposed to correct cursor position

### DISCUSSION THREAD

**Discussion by pbienst** [SilentVoid13/Templater#781] [Help] [answered]
**Plugin information:**

-   OS: Win 11
-   Templater version: 1.12.0
-   Obsidian version: 0.14.6
-   Templater settings: Template folder location: "templates/", automatic jump to cursor on

**Describe the bug**

I have the following template:

```
---
date: <%tp.date.now("YYYY-MM-DD") %>T00:00
tags:
   - planner
---

<% tp.web.daily_quote() %>
<% await tp.file.move("/journal work/" + tp.file.title) %>
<% tp.file.rename(tp.date.now("YYYYMMDD")+ " Planner") %>

# Planned
- [ ] <% tp.file.cursor(1) %>

# Unplanned
*
```

When creating a file from the template, the ``` <% tp.file.cursor(1) %>``` statement gets erased, but the cursor position is actually at the title of the note, with the entire title selected.

**Expected behavior**

The cursor should be at the position of  ``` <% tp.file.cursor(1) %>```

**Comment by liamcain** [comment]
When you press <kbd>Enter</kbd> to finish the rename, is the cursor at the correct position?

**Comment by pbienst** [comment]
If I press enter, it deletes the title. Tab brings the cursor to the correct position. So, it's a matter of the keyboard focus being wrong.

**Comment by liamcain** [comment]
This is by design. "Create new note from template" focuses the title to give you an opportunity to name your new note. (By default, the title is "Untitled 1")

Since you're renaming the template automatically, I suggest adding the following to your template to focus the new note instead of the titlebar:

```js
<%* app.workspace.activeLeaf.view.editor.focus() -%>
```

**Comment by AB1908** [comment]
This will be handy. Thanks Liam.

**Comment by pbienst** [comment]
I'm afraid that option does not seem to work...

**Comment by liamcain** [comment]
You're right, sorry about that.

Here's a version that should work:

```
---
date: <%tp.date.now("YYYY-MM-DD") %>T00:00
tags:
   - planner
---

<% tp.web.daily_quote() %>
<% await tp.file.move("/journal work/" + tp.date.now("YYYYMMDD")+ " Planner") %>
<%* setTimeout(() => app.workspace.activeLeaf.view.editor.focus(), 200) -%>

# Planned
- [ ] <% tp.file.cursor(1) %>

# Unplanned
*
```

**Comment by pbienst** [comment]
Thanks, but still no luck...

**Comment by liamcain** [comment]
Strange, you can try increasing the timeout (so changing the 100 to something higher, like 200 or 300). I'll take a look to see if there's a better way to handle this without adding a hack to the template.

**Comment by pbienst** [comment]
300 still does not work, I'm afraid...

**Comment by Zachatoo** [comment, marked as answer]
I'm adding a "Jump to note content after creating note" setting, when enabled, will jump your cursor to the note content without having to tab into it.

<img width="888" height="776" alt="Screenshot 2026-06-05 at 10 05 48 PM" src="https://github.com/user-attachments/assets/5a5b7074-66b9-4f63-b266-d272141d2d7e" />

This is in the Templater 2.22.0 release, which is only available on Obsidian 1.13.0+
